docs/Connection-Pools.md

```diff
@@ -0,0 +1,110 @@
+
+In this example we create a PostgreSQL connection pool:
+
+```
+auto config = std::make_shared<sqlpp::postgresql::connection_config>();
+config->dbname = "my_database";
+config->user = "my_user";
+config->password = "my_password";
+config->debug = true;
+auto pool = sqlpp::postgresql::connection_pool{config, 5};
+```
+You can also create a pool object without a configuration and initialize it later.
+
+```
+auto pool = sqlpp::postgresql::connection_pool{}
+....
+....
+....
+auto config = std::make_shared<sqlpp::postgresql::connection_config>();
+config->dbname = "my_database";
+config->user = "my_user";
+config->password = "my_password";
+config->debug = true;
+pool.initialize(config, 5);
+```
+## Getting connections from the connection pool
+
+Once the connection pool object is established we can use the _get()_ method to fetch connections
+
+```
+auto db = pool.get();
+for (row : db(select(....))) {
+  ....
+}
+```
+
+## Returning connections to the connection pool
+
+We don't really need to do anything to return the connection to the pool. Once the connection object's destructor is called the connection is not really destroyed but instead is returned automatically to the connection
+pool's cache. This means that we can use the connection pool in the following way
+
+```
+for (row : pool.get()(select(....))) {
+  ....
+}
+```
+In the above example we fetched a connection from the pool, used to make an SQL query and then returned the connection to the pool.
+
+## Ensuring that connections handed out by the connection pool are valid
+
+Connection pools handle out connections that are either newly created or fetched from the connection cache. For connections that are fetched from the cache an optional check can be made to ensure that the connection is still active.
+If the cached connection is no longer active, then it is discarded and the user is handed a newly created connection.
+
+The check type is specified as an optional parameter of the _get()_ method. Currently the following check types are supported:
+
+* **sqlpp::connection_check::none** Don't check the connection
+* **sqlpp::connection_check::passive** A passive check which does not send anything to the server but just checks if the server side has already closed their side of the connection. This check type is supported only for PostgreSQL, for the other connector types it is treated as _none_.
+* **sqlpp::connection_check::ping** Send a dummy request to the server to check if the connection is still alive. For MySQL connections this check uses the `mysql_ping` library function. For the other connector types
+this check sends `SELECT 1` to the server.
+
+For example:
+
+```
+auto db = pool.get(sqlpp::connection_check::ping);
+for (row : db(select(....))) {
+  ....
+}
+```
+
+## Working around connection thread-safety issues
+
+Connection pools can be used to workaround [thread-safety issues](Threads.md) by ensuring that no connection is used simultaneously by multiple threads.
```

Comment:
note: workaround is a noun, not a verb

Reply:
Changed "to workaround" -> "to work around"